src/components/pages/ImageClassification/ImageAnnotationModal/ImageAnnotationModal.js

```diff
@@ -44,6 +47,17 @@ const ImageAnnotationModal = ({
   const [hoveredAttributeId, setHoveredAttributeId] = useState('')
   const [isSaving, setIsSaving] = useState(false)
   const [isDataUpdatedSinceLastSave, setIsDataUpdatedSinceLastSave] = useState(false)
+  const [hasMapLoaded, setHasMapLoaded] = useState(false)
+  const map = useRef(null)
+
+  const { imageScale } = useImageScale({ hasMapLoaded, dataToReview })
+
+  const { getPointsGeojson } = usePointsGeoJson({ dataToReview, imageScale, map })
+
+  const { zoomToPointsByAttributeId } = useZoomToPointsByAttributeId({
+    getPointsGeojson,
+    mapRef: map,
+  })
```

Comment:
These new hooks and state are mostly just me moving stuff out of the IC map component so that we can zoom from the table component. Ill point out code within them that is new if its not just copy paste.